types/validator.go

```diff
@@ -94,6 +97,36 @@ func (v *Validator) Bytes() []byte {
 	})
 }
 
+func (v *Validator) ToProto() (*tmproto.Validator, error) {
+	pk, err := ce.PubKeyToProto(v.PubKey)
+	if err != nil {
+		return nil, err
+	}
+
+	vp := tmproto.Validator{
+		Address:          v.Address,
+		PubKey:           pk,
+		VotingPower:      tmmath.SafeConvertUint64(v.VotingPower),
+		ProposerPriority: v.ProposerPriority,
+	}
+
+	return &vp, nil
+}
+
+func (v *Validator) FromProto(vp tmproto.Validator) error {
+	pk, err := ce.PubKeyFromProto(vp.PubKey)
+	if err != nil {
+		return err
+	}
+
+	v.Address = vp.Address
+	v.PubKey = pk
+	v.VotingPower = tmmath.SafeConvertInt64(vp.VotingPower)
```

Comment:
does that mean anybody will be able to crash our node by sending `vp.VotingPower` > `math.MaxInt64`?

Reply:
undid the proto type change to uint64 for now. this will happen at a later time. https://github.com/tendermint/tendermint/pull/4655